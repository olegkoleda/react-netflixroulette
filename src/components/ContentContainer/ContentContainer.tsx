import React, { SetStateAction, useEffect, useState } from "react";
import Header from "../Header";
import Footer from "../Footer";
import Main from "../Main";
import { Composition } from "atomic-layout";
import { movies as mockData } from "../../movies";

const defaultMovie = {
  id: 181808,
  title: "React App",
  tagline: "The Saga Continues",
  vote_average: 7.1,
  vote_count: 4732,
  release_date: "2017-12-13",
  poster_path:
    "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
  overview:
    "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
  budget: 200000000,
  revenue: 1325937250,
  genres: ["Fantasy", "Adventure", "Science Fiction"],
  runtime: 152,
};

const ContentContainer: React.FC = () => {
  const [movieId, setMovieId] = useState(null);
  const [showDetails, setShowDetails] = useState(false);
  const activeMovie = mockData.find(({ id }) => id === movieId) || defaultMovie;

  useEffect(() => {
    document.title = showDetails ? activeMovie.title : "NetflixRoulette";
  }, [showDetails, activeMovie]);

  const changeMovie = (id: React.SetStateAction<null>) => {
    setMovieId(id);
    setShowDetails(true);
  };

  return (
    <Composition templateRows="50vh 1fr 4.5rem">
      <Header
        showDetails={showDetails}
        movie={activeMovie}
        changeView={setShowDetails}
      />
      <Main movies={mockData} changeMovie={changeMovie} />
      <Footer />
    </Composition>
  );
};

export default ContentContainer;
