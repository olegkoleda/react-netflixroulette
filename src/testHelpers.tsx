import React from "react";
import renderer from "react-test-renderer";
import { theme } from "./styles/Theme";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { initialAppState } from "./store/reducers/rootReducer";

const middleWares = [thunk];
const mockStore = configureMockStore(middleWares);
const store = mockStore(initialAppState);

export function renderWithTheme(component: React.ReactNode) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export const reduxWrapper = (Component: typeof React.Component | React.FC, data = {}) => (
  <Provider store={store}>
    <Router>
      <Component {...data}/>
    </Router>
  </Provider>
);
