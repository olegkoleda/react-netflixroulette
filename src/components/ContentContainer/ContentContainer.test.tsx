import ContentContainer from "./ContentContainer";
import { renderWithTheme, reduxWrapper } from "../../testHelpers";

test("ContentContainer renders correctly", () => {
  const component = renderWithTheme(reduxWrapper(ContentContainer));

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
