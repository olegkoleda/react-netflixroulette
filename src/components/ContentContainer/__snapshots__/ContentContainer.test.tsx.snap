// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer renders correctly 1`] = `
<div
  className="sc-AxjAm beCVJq"
>
  <div
    className="styledheader__StyledHeaderWrapper-ntr33y-2 hcnsvj sc-AxjAm feVghO"
    height="50vh"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt=""
      className="styledheader__StyledHeaderImage-ntr33y-0 hHPvMt"
      src="test-file-stub"
    />
    <div
      className="styledheader__StyledBackground-ntr33y-1 fkOeYX"
    />
    <header
      className="sc-AxjAm iZLCHR"
    >
      <h1>
        <span>
          <span
            className="styledlogo__StyledLogo-sc-1suyno6-0 dKPWah"
            fontWeight={800}
          >
            netflix
          </span>
          <span
            className="styledlogo__StyledLogo-sc-1suyno6-0 eKILPN"
            fontWeight={400}
          >
            roulette
          </span>
        </span>
      </h1>
      <button
        className="styledbutton__StyledButton-sc-7hz9n2-0 jSpEnr"
        onClick={[Function]}
      >
        + ADD MOVIE
      </button>
    </header>
    <div
      className="sc-AxjAm bhDiCM"
      height="100%"
    >
      <div
        className="sc-AxjAm fqTMaV"
        width="100%"
      >
        <h2
          className="Heading-sc-15b2y3d-0 fKqoJe"
        >
          Find your movie
        </h2>
        <form
          action="#"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="sc-AxirZ hhbvJa"
          >
            <input
              className="styledsearch__StyledInput-sc-112aji7-0 gYPiLC"
              name="search"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="What do you want to watch?"
              type="text"
              value=""
            />
            <button
              className="styledbutton__StyledButton-sc-7hz9n2-0 dUEACD"
              type="submit"
            >
              SEARCH
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <main
    className="styledmain__StyledMain-sc-1oh6sq-0 ihRYAc sc-AxjAm cPZRBP"
  >
    <div
      className="sc-AxirZ eCfaaJ"
    >
      <div
        className="sc-AxjAm gDgSnB"
      >
        <div
          className="sc-AxjAm fRMFrE"
          onChange={[Function]}
        >
          <div
            checked={true}
            className="styledradioButton__StyledRadioButton-sc-1athmhv-0 bdPXCN"
          >
            <input
              defaultChecked={true}
              id="filter_All"
              name="filter"
              type="radio"
              value=""
            />
            <label
              htmlFor="filter_All"
            >
              ALL
            </label>
          </div>
          <div
            checked={false}
            className="styledradioButton__StyledRadioButton-sc-1athmhv-0 bNqTgl"
          >
            <input
              defaultChecked={false}
              id="filter_Documentary"
              name="filter"
              type="radio"
              value="Documentary"
            />
            <label
              htmlFor="filter_Documentary"
            >
              DOCUMENTARY
            </label>
          </div>
          <div
            checked={false}
            className="styledradioButton__StyledRadioButton-sc-1athmhv-0 bNqTgl"
          >
            <input
              defaultChecked={false}
              id="filter_Comedy"
              name="filter"
              type="radio"
              value="Comedy"
            />
            <label
              htmlFor="filter_Comedy"
            >
              COMEDY
            </label>
          </div>
          <div
            checked={false}
            className="styledradioButton__StyledRadioButton-sc-1athmhv-0 bNqTgl"
          >
            <input
              defaultChecked={false}
              id="filter_Horror"
              name="filter"
              type="radio"
              value="Horror"
            />
            <label
              htmlFor="filter_Horror"
            >
              HORROR
            </label>
          </div>
        </div>
      </div>
      <div
        className="sc-AxjAm TBlcn"
      >
        <div
          className="styledsort__StyledSelectWrapper-sc-13rug2s-0 cQsuLb sc-AxjAm kDfEX"
        >
          <label
            className="styledsort__StyledLabel-sc-13rug2s-1 iVKDWt"
            htmlFor="sort-by"
          >
            SORT BY
          </label>
          <select
            className="styledsort__StyledSelect-sc-13rug2s-3 iacsnF"
            defaultValue="release_date"
            id="sort-by"
            onChange={[Function]}
          >
            <option
              className="styledsort__StyledOption-sc-13rug2s-2 kiNmiY"
              value="release_date"
            >
              RELEASE DATE
            </option>
            <option
              className="styledsort__StyledOption-sc-13rug2s-2 kiNmiY"
              value="rating"
            >
              RATING
            </option>
            <option
              className="styledsort__StyledOption-sc-13rug2s-2 kiNmiY"
              value="budget"
            >
              BUDGET
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm gNWrut"
    >
      <h3
        className="Heading-sc-15b2y3d-0 fKqoJe"
      >
        No movie found
      </h3>
    </div>
  </main>
  <footer
    className="styledfooter__StyledFooter-vo699h-0 cZSprY sc-AxjAm bqPLoD"
    height="4.5rem"
  >
    <span>
      <span
        className="styledlogo__StyledLogo-sc-1suyno6-0 dKPWah"
        fontWeight={800}
      >
        netflix
      </span>
      <span
        className="styledlogo__StyledLogo-sc-1suyno6-0 eKILPN"
        fontWeight={400}
      >
        roulette
      </span>
    </span>
  </footer>
</div>
`;
